### ISSUE TITLE
JWindow translucency breaks on Linux Mint with 2 screens

### ISSUE THREAD

**Issue by stefan-reich** [adoptium/adoptium-support#430]
## Summary

In a multi-screen setup, in some parts of the screen JWindow translucency doesn't work... [Screenshot.](https://botcompany.de/images/1103064)

## Steps to reproduce

Just run this program:

```
import java.util.List;
import java.util.*;
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;

class TransparencyBug {

public static void main(final String[] args) throws Exception {
  var screens = GraphicsEnvironment.getLocalGraphicsEnvironment().getScreenDevices();
  System.out.println("Detected " + screens.length + " screen(s)");
  for (int iScreen = 0; iScreen < screens.length; iScreen++) {
    var screen = screens[iScreen].getDefaultConfiguration().getBounds();
    
    // make a grid of windows covering each screen
    
    int cols = 12, rows = 8;
    for (Rectangle r : gridOfRectangles(screen, cols, rows)) {
      JWindow window = new JWindow();
      window.setBounds(r);
        
      // faint (mostly translucent) blue background
      window.setBackground(new Color(0, 0, 255, 64));
      
      window.setAlwaysOnTop(true);
      //var label = new JLabel("Screen " + (iScreen+1) + " of " + screens.length);
      //label.setHorizontalAlignment(JLabel.CENTER);
      //window.add(label);
      disposeWindowOnClick(window);
      window.setVisible(true);
    }
  }
    
  System.out.println("Will exit in 10");
  Thread.sleep(10000);
  System.exit(0);
}
static List<Rectangle> gridOfRectangles(Rectangle r, int cols, int rows) {
  List<Rectangle> l = new ArrayList();
  for (int row = 0; row < rows; row++) {
    int y1 = r.y+(int) Math.round(r.height*(double) row/rows);
    int y2 = r.y+(int) Math.round(r.height*(double) (row+1)/rows);
    for (int col = 0; col < cols; col++) {
      int x1 = r.x+(int) Math.round(r.width*(double) col/cols);
      int x2 = r.x+(int) Math.round(r.width*(double) (col+1)/cols);
      l.add(new Rectangle(x1, y1, x2-x1, y2-y1));
    }
  }
  return l;
}

static void disposeWindowOnClick(final Window window) {
  AWTEventListener l = new AWTEventListener() {
    public void eventDispatched(AWTEvent evt) {
      if (!window.isVisible()) {
        //print("disposeWindowOnClick: Forgetting listener");
        Toolkit.getDefaultToolkit().removeAWTEventListener(this);
        return;
      }
        
      if (!(evt instanceof MouseEvent)) return;
      Object source = evt.getSource();
      if (source == window && evt.getID() == MouseEvent.MOUSE_PRESSED) {
        Toolkit.getDefaultToolkit().removeAWTEventListener(this);
        window.dispose();
      }
    }
  };
  Toolkit.getDefaultToolkit().addAWTEventListener(l, AWTEvent.MOUSE_EVENT_MASK);
}
}
```

## Expected results

It shows a grid of JWindows covering all screens. The windows are supposed to be mostly translucent.

## Actual results

In some parts of the screen, they are not translucent. [Screenshot.](https://botcompany.de/images/1103064)

Additionally, for these JWindows, task bar icons appear.

## Triaging info

Java version:

openjdk version "17.0.1" 2021-10-19
OpenJDK Runtime Environment Temurin-17.0.1+12 (build 17.0.1+12)
OpenJDK 64-Bit Server VM Temurin-17.0.1+12 (build 17.0.1+12, mixed mode, sharing)

What is your operating system and platform?

Linux Mint 19. Maybe I should change something in CompizConfig? I think I am back to Linux Mint's default settings there.

How did you install Java?

From tar.gz

Did it work before?

No

Did you test with other Java versions?

No, not yet

**Comment by karianna** [comment]
Hi there, can you try a nightly build at: https://adoptium.net/nightly

**Comment by stefan-reich** [comment]
Is there another group to post this? Someone dedicated to AWT internals?

**Comment by karianna** [comment]
yes the openjdk client-libs-dev mailing list is probably your next step

**Comment by stefan-reich** [comment]
Perfect thank you. This really bothers me. I need those overlays. Thanks

**Comment by jerboaa** [comment]
Upstream bug: https://bugs.openjdk.java.net/browse/JDK-8280482

**Comment by karianna** [comment]
Upstream bug is still open

**Comment by stefan-reich** [comment]
True. Apparently not a very high priority bug at Oracle

**Comment by karianna** [comment]
Upstream bug is still open

**Comment by adamish** [comment]
I think I might be observing something similar...

Details on https://stackoverflow.com/questions/75807131/java-swing-window-looses-transparency-on-secondary-monitors-in-linux

TL;DR... Swing correctly selects a 32 bit X11 visual to support transparency but this gets changed to the default X11 visual which is 24 bits when the window is moved.

**Comment by stefan-reich** [comment]
@adamish That sounds excellent, looks like you found the source of the problem!

**Comment by adamish** [comment]
@stefan-reich I'd be interested to know if your issue is same root cause. It's a complex area, there could be more than one fault. I'll see if I can get openJDK access to post my findings to upstream https://bugs.openjdk.java.net/browse/JDK-8280482

FYI I came up with another patch using a different approach

```
AtomicBoolean updating = new AtomicBoolean();
frame.addPropertyChangeListener("graphicsConfiguration", new PropertyChangeListener() {
    @Override
    public void propertyChange(PropertyChangeEvent evt) {
        if (updating.compareAndSet(false, true)) {
            /*
             * trigger frame to pick a graphics context with transparency support again
             */
            try {
                frame.setBackground(new Color(0, 0, 0, 255));
                frame.setBackground(new Color(0, 0, 0, 0));
            } finally {
                updating.set(false);
            }
        }
    }
});
```

**Comment by karianna** [comment]
@adamish LMK if you get stuck and I'll post your comment in the bug

**Comment by adamish** [comment]
@karianna Yes please, seems getting access to openjdk jira is quite a difficult process.

**Comment by karianna** [comment]
Added
